lib/queryContextCache.js

```diff
@@ -0,0 +1,391 @@
+/*
+ * Copyright (c) 2012-2022 Snowflake Computing Inc. All rights reserved.
+ */
+const Logger = require('./logger');
+const crypto = require('crypto');
+/**
+     * Constructor.
+     *
+     * @param id database id
+     * @param timestamp Server time when this entry read
+     * @param priority Priority of this entry w.r.t other ids
+     * @param context Opaque query context, used by query processor in the server.
+     */
+
+/** Query context information. */
+function QueryContextElement (id,timestamp,priority,context) {
+  this.id = id;
+  this.timestamp = timestamp;
+  this.priority = priority;
+  this.context = context;
+}
+
+QueryContextElement.prototype.getId = function () {
+  return this.id;
+};
+
+QueryContextElement.prototype.getReadTimestamp = function () {
+  return this.timestamp;
+};
+
+QueryContextElement.prototype.getPriority = function () {
+  return this.priority;
+};
+
+QueryContextElement.prototype.getContext = function () {
+  return this.context;
+};
+ 
+QueryContextElement.prototype.setId = function (id) {
+  this.id = id;
+};
+
+QueryContextElement.prototype.setReadTimestamp = function (timestamp) {
+  this.timestamp = timestamp;
+};
+
+QueryContextElement.prototype.setPriority = function (priority) {
+  this.priority = priority;
+};
+
+QueryContextElement.prototype.setContext = function (context) {
+  this.context = context;
+};
+
+QueryContextElement.prototype.equals = function (obj) {
```

Comment:
do we use equals and hashCode methods here?

Reply:
we can add them in the future but for now they are not necessary - Javascript structures won't use them as Java/.Net Maps?

Reply:
I will double-check. If those are not necessary, I will remove them.